usersession/agent/rest_api.go

```diff
@@ -290,15 +292,15 @@ func postPendingRefreshNotification(c *Command, r *http.Request) Response {
 
 	msg := &notification.Message{
 		AppName: refreshInfo.BusyAppName,
-		Summary: summary,
+		Title:   summary,
 		Icon:    icon,
 		Body:    body,
 		Hints:   hints,
 	}
 
 	// TODO: silently ignore error returned when the notification server does not exist.
 	// TODO: track returned notification ID and respond to actions, if supported.
-	if _, err := notifySrv.SendNotification(msg); err != nil {
+	if err := notifySrv.SendNotification(notification.ID(refreshInfo.InstanceName), msg); err != nil {
```

Comment:
Maybe I'm misunderstanding the context here, but it would seem more logical to me if the notification manager were instantiated with the ID `refreshInfo.InstanceName`, and the notification were sent with a fixed `snap-refresh` ID (though, are we sure we'll always send just one notification per snap for the whole duration of the session? What if the snap gets refreshed twice? I guess we shouldn't reuse the ID, but increment it). So that we might support sending more notifications of different types for the same snap.

Reply:
yeah I wonder if this ID should be the instance name + the revision number, it does seem like this code as-is will use the same notification client ID for all refreshes of the same snap

Reply:
or actually perhaps it's fine because we call "CloseNotification" after the refresh is successfully performed, in which case the client ID for the snap instance name is forgotten and creating a new notification for a second refresh will just create a fresh new notification ?

Reply:
The desktop ID passed to `NewNotificationManager` identifies the source of the notifications, which in this case is the session agent.

We explicitly don't want to forge the notifications to appear to come from the snap application, for a number or reasons:

1. When using the GTK implementation, the desktop ID also identifies the D-Bus name that notification actions/activations will be sent to. That should be to the session agent rather than the app.
2. With the GTK backend's client-chosen notification IDs, we have no idea what IDs the snap application is using.
3. Desktops may group notifications from a single source together, or let users configure how notifications from a particular source are displayed. Making our notifications appear to come from different sources would be user hostile.

In the case of GNOME, we currently have the following notification preferences exposed in the control center:

![image](https://user-images.githubusercontent.com/12724165/135223701-26ad0637-0fbb-4728-be17-ddca2fa388e3.png)

As far as the actual notification ID goes, I would suggest prefixing it with something (maybe use 'refresh:$SNAP_INSTANCE_NAME`?). It is quite possible that the session agent will want to post more types of notifications in future, so namespacing the IDs will help avoid conflicts.

Reply:
Thanks for these remarks. I like the "refresh:$SNAP_INSTANCE_NAME" idea 👍